src/app/mod.rs

```diff
@@ -43,7 +42,8 @@ use tonic::{Request, Response, Status};
 use tracing::{debug, info};
 
 type MainStore<S> = SharedStore<RevertibleStore<S>>;
-type ModuleStore<S> = SubStore<MainStore<S>>;
+type ModuleStore<S> = RevertibleStore<S>;
+type ModuleList<S> = Vec<(Identifier, Box<dyn Module<ModuleStore<S>>>)>;
 type Shared<T> = Arc<RwLock<T>>;
```

Comment:
Maybe a better name here is `SharedRw<T>`? To me, `Shared` would be equivalent to what `Arc` provides.

Reply:
Great suggestion! 💯 I think I agree with this and most other comments in this review. Will be addressing them soon.

Reply:
https://github.com/informalsystems/basecoin-rs/pull/44/commits/3c2e55eb1b57115cf64df4da4ddfbd0f3184f129